app/src/github.com/harithj/ci-cd-2019/github/git/git.go

```diff
@@ -0,0 +1,27 @@
+package git
+
+import (
+  "os"
+  "fmt"
+
+  "gopkg.in/src-d/go-git.v4"
+  "gopkg.in/src-d/go-git.v4/plumbing"
+
+  "github.com/harithj/ci-cd-2019/github/payload"
+)
+
+func CloneRepo(payload *payload.Payload) string {
+  cloneDir := "/temp/" + payload.RepoName
+  os.Mkdir(cloneDir, 0777)
```

Comment:
this function returns an error, you should include proper error handling for easier debugging in future.
```
        if err := os.Mkdir(cloneDir, 0777); err != nil {
		...
	}
```